From 059d0c86ad6231680b19bf0eb5b572fbdff5f0db Mon Sep 17 00:00:00 2001
From: Nicola Murino <nicola.murino@gmail.com>
Date: Wed, 30 Oct 2019 14:59:40 +0100
Subject: [PATCH 33/45] glimagesink: fix upper-left and upper-right rotate
 matrices

(cherry picked from commit 4ed345b7ebc1992d7013cff1a1dcd1814ee9c69b)
Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 ext/gl/gstglimagesink.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/ext/gl/gstglimagesink.c b/ext/gl/gstglimagesink.c
index 4a7f70f..0de3e5f 100644
--- a/ext/gl/gstglimagesink.c
+++ b/ext/gl/gstglimagesink.c
@@ -486,16 +486,16 @@ static const gfloat vertical_flip_matrix[] = {
 
 /* upper-left-diagonal */
 static const gfloat upper_left_matrix[] = {
-  0.0f, 1.0f, 0.0f, 0.0f,
-  1.0f, 0.0f, 0.0f, 0.0f,
+  0.0f, -1.0f, 0.0f, 0.0f,
+  -1.0f, 0.0f, 0.0f, 0.0f,
   0.0f, 0.0f, 1.0f, 0.0f,
   0.0f, 0.0f, 0.0f, 1.0f,
 };
 
 /* upper-right-diagonal */
 static const gfloat upper_right_matrix[] = {
-  0.0f, -1.0f, 0.0f, 0.0f,
-  -1.0f, 0.0f, 0.0f, 0.0f,
+  0.0f, 1.0f, 0.0f, 0.0f,
+  1.0f, 0.0f, 0.0f, 0.0f,
   0.0f, 0.0f, 1.0f, 0.0f,
   0.0f, 0.0f, 0.0f, 1.0f,
 };
-- 
2.20.1

